core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -114,4 +116,20 @@ private static TableMetadata findFirstExistentPreviousMetadata(List<MetadataLogE
     }
     return manifestListLocations;
   }
+
+  /**
+   * Returns locations of manifest lists in a table.
+   *
+   * @param table table for which manifestList needs to be fetched
+   * @param snapshotIds snpashots to filter for
+   * @return the location of manifest Lists
+   */
+  public static List<String> manifestListLocations(Table table, Set<Long> snapshotIds) {
+    Iterable<Snapshot> snapshots = table.snapshots();
+    return StreamSupport.stream(snapshots.spliterator(), false)
```

Comment:
Can you add a comment to clarify the `false` argument?

Reply:
Done